src/Confluent.Kafka/Serialization/StringSerializer.cs

```diff
@@ -33,25 +38,88 @@ public class StringSerializer : ISerializer<string>
         /// </param>
         public StringSerializer(Encoding encoding)
         {
-            this.encoding = encoding;
+            this.keyEncoding = encoding;
+            this.valueEncoding = encoding;
+        }
+
+        private static Encoding StringToEncoding(string s) 
+        {
+            switch (s)
+            {
+                case "UTF8":
+                    return Encoding.UTF8;
+                case "UTF7":
+                    return Encoding.UTF7;
+                case "UTF32":
+                    return Encoding.UTF32;
+                case "ASCII":
+                    return Encoding.ASCII;
+                case "Unicode":
+                    return Encoding.Unicode;
+                case "BigEndignUnicode":
+                    return Encoding.BigEndianUnicode;
+                default:
+                    throw new ArgumentException("unknown string encoding: " + s);
+            }
+        }
+
+        /// <summary>
+        ///     Initializes a new StringSerializer class instance.
+        /// </summary>
+        /// <param name="config">
+        ///     Configuration properties.
+        /// </param>
+        public StringSerializer(IEnumerable<KeyValuePair<string, object>> config)
+        {
+            if (config.Count(ci => ci.Key == "string.serializer.encoding.key") > 0)
```

Comment:
Unfortunately the implementation is a bit ugly - the object is unaware of it's context when constructed, so it needs to be prepared for use with either keys or values, even though this is known at construction time and the serializer instance will only be ever used for one or the other.